### PR TITLE
feat: ask user to switch to strength search if connection can be established while using map search

### PR DESCRIPTION
**Pull request by NicoStellwag**
In collab with @domi1504

### CODE REVIEW

File: Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs [hunk 1]
```diff
@@ -8,31 +8,35 @@
...
 {
     public class MapViewModel : BaseViewModel
     {
+        public bool DeviceNotNull => Device != null;
+
         private readonly Xamarin.Forms.Maps.Map map;
         private readonly IGeolocation geolocation;
+        private readonly INavigator navigator;
         private readonly IDevicesStore devicesStore;
+        private bool userDeclinedPageSwitch, showingDialogue;
```
Comment (domi1504) [inline Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs:27]: `userDeclinedPageSwitch` is not necessary anymore, is it?

**Review by domi1504 — APPROVED** [review]
Apart from the two tiny comments it looks fine to me.

File: Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs (hunk 1 above)
Comment (NicoStellwag) [reply Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs:27]: nice catch, the userdeclined... can be removed, I simply forgot that.

File: Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs [hunk 2]
```diff
@@ -88,7 +125,16 @@ public async void OnAppearing()
...
+
+            userDeclinedPageSwitch = false;
+            showingDialogue = false;
+            devicesStore.DevicesChanged += CheckIfSelectedDeviceReachable;
+
             ShowSelectedDevice();
         }
+
+        public void OnDisappearing() {
+            devicesStore.DevicesChanged -= CheckIfSelectedDeviceReachable;
```
[the remark this answers is not in the document]
Comment (NicoStellwag) [reply Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/MapViewModel.cs:137]: should not be a problem:
https://stackoverflow.com/questions/35407570/c-sharp-should-i-prevent-an-event-to-be-unhooked-twice

**Review by Interperle — APPROVED** [review]
Looks fine with Domis first change

**Pull request merged**